### PR TITLE
Tweak the basic subpass dependencies

### LINKED ISSUE #665: Misleading example base subpass dependencies?
```

	dependencies[0].srcSubpass = VK_SUBPASS_EXTERNAL;
	dependencies[0].dstSubpass = 0;
	dependencies[0].srcStageMask = VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
	dependencies[0].dstStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
	dependencies[0].srcAccessMask = VK_ACCESS_MEMORY_READ_BIT;
	dependencies[0].dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
	dependencies[0].dependencyFlags = VK_DEPENDENCY_BY_REGION_BIT;
```

Not entirelly sure why exacly it is `VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT` (blocking _execution dependency_), so it probably needs a comment. Probably it should be either `ALL_COMMANDS` and have `WRITE` too, or it should be `TOP_OF_PIPE`  and the access mask should be `0`.

Furthemore, the `loadOp` is set as `CLEAR`, so only `VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT` should be needed.

```
	dependencies[1].srcSubpass = 0;
	dependencies[1].dstSubpass = VK_SUBPASS_EXTERNAL;
	dependencies[1].srcStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
	dependencies[1].dstStageMask = VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
	dependencies[1].srcAccessMask = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
	dependencies[1].dstAccessMask = VK_ACCESS_MEMORY_READ_BIT;
	dependencies[1].dependencyFlags = VK_DEPENDENCY_BY_REGION_BIT;
```

`VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT` has no memory access. The wildcard `VK_ACCESS_MEMORY_READ_BIT` is technically allowed, though the access mask should probably be `0`

Furthermore the StoreOp is `STORE`, so only `VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT` should suffice.

### PR DESCRIPTION
**Pull request by krOoze**
fix #665

### CODE REVIEW

File: base/vulkanexamplebase.cpp [hunk 1]
```diff
@@ -2145,18 +2145,20 @@ void VulkanExampleBase::setupRenderPass()
 
 	dependencies[0].srcSubpass = VK_SUBPASS_EXTERNAL;
 	dependencies[0].dstSubpass = 0;
-	dependencies[0].srcStageMask = VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
+	// cannot be much sure here what the previous work is
+	// assumably it is the semaphore to vkQueueSubmit, otherwisely it is the user's resposibility to otherwisely synchronize
+	dependencies[0].srcStageMask = this->submitPipelineStages;
```
Comment (krOoze) [inline base/vulkanexamplebase.cpp]: Too many examples rely on this, so I am not 110 % sure what the correct stage should be here...

**Comment by SaschaWillems** [comment]
Thanks for your PR. I'll try to take a look at this as soon as possible, but it may take some time.

**Comment by SaschaWillems** [comment]
LGTM so far. Will do a bit of additional testing before merging.

**Comment by SaschaWillems** [comment]
Sorry for the long delay, but haven't had time to test this earlier. I checked with all samples, and except for one this works as expected. The only sample no longer working properly is the pipeline statistics one, so before merging I would need to fix that sample to e.g. create it's own render pass.

Will need some additional time to test this though, as this may have potential side effects except for the one obviously visible in the pipeline statistics sample.

**Comment by krOoze** [comment]
Right, I faced the same problem. Sync is fundamentally untestable. And I don't really want to inspect each of the 50 examples manually.

Let's perhaps start with just the Triangle example first: #681.

**Comment by SaschaWillems** [comment]
I did an initial pass to fix the basic dependencies as per discussion in #665.

I hope it's okay if I close this PR then.

**Pull request closed without merging**